src/components/PopupWithConfirmation.js

```diff
@@ -0,0 +1,29 @@
+import Popup from "./Popup.js";
+
+export default class PopupWithConfirmation extends Popup {
+  constructor({ popupSelector }) {
+    super({ popupSelector });
+    this._modalForm = this._popupElement.querySelector(".modal__form");
+    this._submitButton = this._popupElement.querySelector(".modal__button");
+  }
+
+  setSubmitText(submit, submitText = "Saving...") {
+    if (submit) {
+      this._submitButton.textContent = submitText;
+    } else {
+      this._submitButton.textContent = this._submitButtonText;
+    }
+  }
+
+  confirmDelete(confirmation) {
```

Comment:
Great job implementing confirmation popup and the functionality related to it! In general, it can support any confirmation.